src/utils/useIsomorphicLayoutEffect.js

```diff
@@ -9,9 +9,16 @@ import { useEffect, useLayoutEffect } from 'react'
 // is created synchronously, otherwise a store update may occur before the
 // subscription is created and an inconsistent state may be observed
 
-export const useIsomorphicLayoutEffect =
+export let useIsomorphicLayoutEffect =
   typeof window !== 'undefined' &&
   typeof window.document !== 'undefined' &&
   typeof window.document.createElement !== 'undefined'
     ? useLayoutEffect
     : useEffect
+
+export const forceUseLayoutEffect = () => {
+  useIsomorphicLayoutEffect = useLayoutEffect
+}
+
+// Supply a getter just to skip dealing with ESM bindings
+export const getUseIsomorphicLayoutEffect = () => useIsomorphicLayoutEffect
```

Comment:
This shouldn't be needed. While imported values are treated as read-only, exported values can change within the module. So, calling `forceUseLayoutEffect()` should be fine for the entirety of the execution context. 

I would add a `forceUseEffect()` as well to allow setting it back.

Reply:
I was under the impression that exported values should change in the importer when changed within the module, but I didn't see that behavior in testing - perhaps an issue in babel? The `getBatch` getter exported from `utils/reactBatchedUpdates.js` was provided for that exact reason (according to @markerikson).

Consider the following test cases. In each, the module is loaded in a beforeEach block with no window.document.createElement (thus setting `useIsomorphicLayoutEffect` to `useEffect`). Calling `forceUseLayoutEffect` after the module is imported, `useIsomorphicLayoutEffect` _is_ `useEffect` unless 1.) the module is re-imported or 2. The getter is used. 
this works as expected:
```
      it('forces useIsomorphicLayoutEffect to useLayoutEffect', async () => {
        forceUseLayoutEffect()
        ;({ useIsomorphicLayoutEffect } = await import(
          '../../src/utils/useIsomorphicLayoutEffect'
        ))

        expect(useIsomorphicLayoutEffect).toBe(useLayoutEffect)
```
but  fails if the re-import is omitted:
```
      it('forces useIsomorphicLayoutEffect to useLayoutEffect', async () => {
        forceUseLayoutEffect()
        //;({ useIsomorphicLayoutEffect } = await import(
        //  '../../src/utils/useIsomorphicLayoutEffect'
        //))

        expect(useIsomorphicLayoutEffect).toBe(useLayoutEffect)
```
->
```Expected: [Function useLayoutEffect]
    Received: [Function useEffect]
```

whereas it passes with the getter:
```
      it('forces useIsomorphicLayoutEffect to useLayoutEffect', async () => {
        forceUseLayoutEffect()
       
        expect(getUseIsomorphicLayoutEffect()).toBe(useLayoutEffect)
      })
```
The same behavior was observed when using the "static" `import { useIsomorphicLayoutEffect } from ...`